meteor/lib/api/rest.ts

```diff
@@ -0,0 +1,204 @@
+import { ClientAPI } from '../api/client'
+import { MethodContext } from './methods'
+import {
+	AdLibActionId,
+	BucketAdLibId,
+	PartId,
+	PieceId,
+	RundownBaselineAdLibActionId,
+	RundownPlaylistId,
+	SegmentId,
+	StudioId,
+} from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { Meteor } from 'meteor/meteor'
+
+export interface RestAPI extends MethodContext {
+	/**
+	 * Returns the current version of Sofie
+	 */
+	index(): Promise<ClientAPI.ClientResponse<{ version: string }>>
+	/**
+	 * Activates a Playlist.
+	 *
+	 * Throws if there is already an active Playlist for the studio that the Playlist belongs to.
+	 * @param rundownPlaylistId Playlist to activate.
+	 * @param rehearsal Whether to activate into rehearsal mode.
+	 * @param connection Connection data including client and header details
+	 */
+	activate(
+		rundownPlaylistId: RundownPlaylistId,
+		rehearsal: boolean,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>>
+	/**
+	 * Deactivates a Playlist.
+	 *
+	 * Throws if the Playlist is not currently active.
+	 * @param rundownPlaylistId Playlist to deactivate.
+	 * @param connection Connection data including client and header details
+	 */
+	deactivate(
+		rundownPlaylistId: RundownPlaylistId,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>>
+	/**
+	 * Executes the requested action by passing `actionId` directly to the blueprints `executeAction` function.
+	 *
+	 * Throws if the target Playlist is not currently active.
+	 * Throws if there is not an on-air Part.
+	 * Throws if the `executeAction` method of the blueprints throws - blueprint error will be returned to caller.
+	 * @returns An object in the form {queuedPartInstanceId?: PartInstanceId, taken?: boolean} (if a part was queued by the action/if the next part was automatically taken)
+	 * @param rundownPlaylistId Playlist to execute action in.
+	 * @param actionId Action Id string, should match something inside the blueprints.
+	 * @param userData Any value, recommended to be a relatively simple { [key: string]: string }-style object, as the object may need to survive a round of serialization.
+	 * @param connection Connection data including client and header details
+	 */
+	executeAction(
+		rundownPlaylistId: RundownPlaylistId,
+		actionId: string,
+		userData: any,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<object>>
```

Comment:
This feels unnecessary? Since there already is an `executeAdLib` method, that accurately allows executing all AdLibs in the system, why is this one needed? How would a system even know what actionId`s are available?

Reply:
I believe the intention here is to be able to execute a 'known' action without it being an adlib in sofie. Allowing the external system to run a super without needing to ingest it into the rundown or a bucket first.

Maybe the name of this method could be changed to clarify that? Im not sure what to

Reply:
This method should be removed I think, it was implemented to reduce the amount of knowledge sharing via the live status gateway required short-term. But the use of it here is that we have separately codified all of the avaialble adlib actions in the blueprints, so an external system can call `executeAction` in the blueprints with arbitrary data. Highly unsafe, definitely should go, but I think it needs to be replaced with a specifiaction for communicating the editable fields of an adlib action so that the other adlib method can be called with arbitrary (within reason) arguments.

Reply:
> Highly unsafe, definitely should go, but I think it needs to be replaced with a specifiaction for communicating the editable fields of an adlib action so that the other adlib method can be called with arbitrary (within reason) arguments.

I suppose for the use-case you are describing, some form of identifying AdLibs, on a machine-readable level would be good. Expose AdLib tags maybe? Or externalId?

Reply:
Aye, I think a combination of exposing `externalId` for actions as well as a structure inside of `userDataManifest.editableFields` that is readable by external systems (perhaps json-schema or adaptive cards) would get us most of the way. We'd then need the `executeAdLib` method to be able to receive data for these editable fields, but we then stand a chance of validating calls.

This sounds like a follow-up discussion/issue, due to potential changes to `userDataManifest.editableFields` having an affect on existing Sofie installations.